### PR TITLE
[13.0][IMP]product_attribute_set: Improve write on category

### PR DESCRIPTION
**Pull request by GuillemCForgeFlow**
Adds the option of writing on many categories at the same time.
cc @ForgeFlow

### CODE REVIEW

**Review by JordiMForgeFlow — APPROVED** [review]
LGTM 👍🏼 , not sure why precommit fails, seems to be something with the configuration. Maybe https://github.com/OCA/odoo-pim/blob/25b1276ef3ca8f77eb4bea7bc3f2e263b74f26ae/.pre-commit-config.yaml#L108

**Review by DavidJForgeFlow — APPROVED** [review]
Functional review :+1: Code also LGTM!

File: product_attribute_set/models/product_category.py [hunk 1]
```diff
@@ -16,12 +16,11 @@ class ProductCategory(models.Model):
...
+        res = super().write(vals)
         if vals.get("attribute_set_id"):
-            product_ids = self.env["product.template"].search(
-                [("categ_id", "=", self.id), ("attribute_set_id", "=", False)]
-            )
-            for product_id in product_ids:
-                product_id.attribute_set_id = self.attribute_set_id
-
-        return True
+            for category in self:
```
Comment (rousseldenis) [inline product_attribute_set/models/product_category.py:21]:
```suggestion
            template_ids = self.env["product.template"].search([
                ("categ_id", "in", self.ids), ("attribute_set_id", "=", False)
            ])
            for category in self:
                template_ids.filtered(lambda t: t.categ_id == category).write({"attribute_set_id": category.attribute_set_id.id})
```

Comment (rousseldenis) [reply product_attribute_set/models/product_category.py:21]: @GuillemCForgeFlow Prefer doing that in order to avoid too much searches in loop

Comment (GuillemCForgeFlow) [reply product_attribute_set/models/product_category.py:21]: I've applied the changes you were suggesting!

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by GuillemCForgeFlow** [comment]
Hi @rousseldenis can we reopen this and get it merged? It's a minor improvement 😄

**Review by rousseldenis — APPROVED** [review]
Seems great. It should be great to add tests in this module.

**Comment by GuillemCForgeFlow** [comment]
> Seems great. It should be great to add tests in this module.

Sure, I'll add them and let you know when it is ready to merge !

**Comment by GuillemCForgeFlow** [comment]
> > Seems great. It should be great to add tests in this module.
> 
> Sure, I'll add them and let you know when it is ready to merge !

@rousseldenis done! 👍🏽

**Review by rousseldenis — APPROVED** [review]

**Comment by rousseldenis** [comment]
@GuillemCForgeFlow Have you seen that this has been adressed differently in v14 ?

https://github.com/OCA/odoo-pim/blob/14.0/product_attribute_set/models/product_category.py

@PierrickBrun

**Comment by GuillemCForgeFlow** [comment]
> @GuillemCForgeFlow Have you seen that this has been adressed differently in v14 ?
> 
> https://github.com/OCA/odoo-pim/blob/14.0/product_attribute_set/models/product_category.py
> 
> @PierrickBrun

I see, but what I'm improving here also is writing on multiple products with one call (not done in v14). Not only the fact that the write method accepts a multi-record call.

If you both agree, maybe we should merge this and forward port these changes to the following versions.

WDYT?
@rousseldenis @PierrickBrun

**Comment by PierrickBrun** [comment]
> If you both agree, maybe we should merge this and forward port these changes to the following versions.
> 
> WDYT? @rousseldenis @PierrickBrun

It looks like the ideal solution, indeed

**Comment by rousseldenis** [comment]
@GuillemCForgeFlow You take care of forwardports ?

**Comment by GuillemCForgeFlow** [comment]
> @GuillemCForgeFlow You take care of forwardports ?

Sure!

**Pull request merged**